WcaOnRails/app/controllers/api/v0/users_controller.rb

```diff
@@ -29,37 +23,43 @@ def show_user_by_wca_id
   end
 
   def permissions
-    if current_user
-      if stale?(current_user)
-        render json: current_user.permissions
-      end
-    else
-      render status: :unauthorized, json: { error: I18n.t('api.login_message') }
+    require_user!
+    if stale?(current_user)
+      render json: current_user.permissions
     end
   end
 
+  def personal_records
+    require_user!
+    return render json: { single: [], average: [] } unless current_user.wca_id.present?
+    person = Person.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(current_user.wca_id)
+    render json: { single: person.ranksSingle, average: person.ranksAverage }
+  end
+
+  def preferred_events
+    require_user!
+    preferred_events = Rails.cache.fetch("#{current_user.id}-preferred", expires_in: 24.hours) do
+      current_user.preferred_events.pluck(:id)
+    end
+    render json: preferred_events
+  end
+
   def bookmarked_competitions
-    if current_user
-      bookmarked_competitions = Rails.cache.fetch("#{current_user.id}-bookmarked", expires_in: 60.minutes) do
-        current_user.competitions_bookmarked.pluck(:competition_id)
-      end
-      render json: bookmarked_competitions
-    else
-      render status: :unauthorized, json: { error: I18n.t('api.login_message') }
+    require_user!
+    bookmarked_competitions = Rails.cache.fetch("#{current_user.id}-bookmarked", expires_in: 60.minutes) do
+      current_user.competitions_bookmarked.pluck(:competition_id)
     end
+    render json: bookmarked_competitions
   end
 
   def token
-    if current_user
-      render json: { status: "ok" }
-    else
-      render status: :unauthorized, json: { error: I18n.t('api.login_message') }
-    end
+    require_user!
+    render json: { status: "ok" }
   end
 
   private
 
-    def show_user(user, show_rankings: false)
+    def show_user(user)
```

Comment:
Please do not remove the `show_rankings` option to maintain backwards compatibility.

Reply:
I added this like 2 mohts ago so I don't expect anyone to actually use it, but I'll keep it in